projects/manage-supervision-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/overview/entity/Requirement.kt

```diff
@@ -157,6 +157,16 @@ interface RequirementRepository : JpaRepository<Requirement, Long> {
         """, nativeQuery = true
     )
     fun getRequirements(id: Long, eventNumber: String): List<RequirementDetails>
+
+    @Query(
+        "SELECT SUM(r.length) " +
+            "FROM Requirement r " +
+            "JOIN  r.mainCategory mc " +
+            "JOIN  r.disposal " +
+            "WHERE r.disposal.id = :id " +
+            "AND mc.code = 'W'"
+    )
```

Comment:
We should use multi-line strings for these queries:
```suggestion
    @Query("""
        SELECT SUM(r.length)
        FROM Requirement r
        JOIN  r.mainCategory mc
        JOIN  r.disposal
        WHERE r.disposal.id = :id
        AND mc.code = 'W'
    """)
```

Reply:
updated